app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,36 @@
+class VotesController < ApplicationController
+  def create
+    if session[:user_id]
+      user = User.find_by(id: session[:user_id])
+      work = Work.find_by(id: params[:work_id])
+      user_votes = user.votes
+      work_votes = work.votes
+
+      user_votes.each do |vote|
+        if work_votes.include?(vote)
+          flash[:status] = :error
+          flash[:message] = "You have already voted for this media"
+          redirect_back(fallback_location: root_path)
+          return
+        end
+      end
+    
+      vote = Vote.new(
+        work_id: params[:work_id],
+        user_id: session[:user_id],
+      )
+      
+      if vote.save
+        flash[:success] = "Successfully upvoted!"
+        redirect_back(fallback_location: root_path)
+      
+        user.votes << vote
+        work.votes << vote
+      end
+
+    else
+      flash[:error] = "A problem occurred: You must login to do that"
+      redirect_back(fallback_location: root_path)
```

Comment:
Typically we put error handling like this at the top of the method:

```ruby
def create
  unless session[:user_id]
    flash[:error] = "A problem occurred: You must login to do that"
    redirect_back(fallback_location: root_path)
    return
  end
  # ... rest of the method ...
end
```

We call this style a "guard clause". This doesn't change the way the code works, but it has two style benefits: it allows you to group all your error checking and handling at the top of the method, so the reader can skim through it and then get to the core functionality, and it also allows you to un-indent everything that's not error handling by one level. The later effect becomes especially important when the error handling starts to get complex.